controller/commentsController.js

```diff
@@ -7,6 +7,63 @@ const CustomError = require("../utils/customError");
 // const User = require("../models/users");
 const responseHandler = require("../utils/responseHandler");
 
+exports.upvoteComment = async (req, res, next) => {
+  try {
+    const { commentId } = req.params;
+    const { ownerId } = req.body;
+
+    if (!mongoose.Types.ObjectId.isValid(commentId)) {
+      next(new CustomError(422, "invalid ID"));
+      return;
+    }
+    const comment = await Comments.findById({ _id: commentId });
+
+    //if user exists in downvotes array
+    if (comment.downVotes.includes(ownerId)) {
+      //get index of user in downvotes array
+      const voterIndex = comment.downVotes.indexOf(ownerId);
+      //if index exists
+      if (voterIndex > -1) {
+        //delete that index
+        comment.upVotes.splice(voterIndex, 1);
```

Comment:
You are splicing the wrong array

```suggestion
        comment.downVotes.splice(voterIndex, 1);
```